client/api_bucket.go

```diff
@@ -333,6 +334,16 @@ func (c *Client) UpdateBucketInfo(ctx context.Context, bucketName string, opts t
 	return c.sendTxn(ctx, updateBucketMsg, opts.TxOpts)
 }
 
+func (c *Client) ToggleSPAsDelegatedAgent(ctx context.Context, bucketName string, opt types.UpdateBucketOptions,
+) (string, error) {
+	_, err := c.HeadBucket(ctx, bucketName)
+	if err != nil {
+		return "", err
+	}
+	msg := storageTypes.NewMsgToggleSPAsDelegatedAgent(c.MustGetDefaultAccount().GetAddress(), bucketName)
```

Comment:
"Toggle pattern" usually expect users clearly know what is the current status so that they can send "toggle" request to set the status to another.  In this way, a query status API usually is needed. 

Why not just using NewMsgSetSPAsDelegateAgent(account, bucketname, true/false)